ddtrace/context.py

```diff
@@ -74,6 +88,63 @@ def _update_tags(self, span):
             span.meta.update(self._meta)
             span.metrics.update(self._metrics)
 
+    def _update_upstream_services(
+        self,
+        span,  # type: Span
+        sampling_priority,  # type: int
+        sampling_mechanism,  # type: SamplingMechanism
+        sample_rate=None,  # type: Optional[float]
+    ):
+        # type: (...) -> None
+        """
+        Update the ``_dd.p.upstream_services`` tag associated with this context.
+
+        Calling with the same values multiple times will not change anything.
+
+        Calling multiple times with new values will replace the previous entry with the new one.
+        (We will only have 1 entry in the tag for this context)
+        """
+
+        entry = "|".join(
+            [
+                # base64 encode binary representation of the service name
+                # DEV: remove padding
+                ensure_text(base64.b64encode(ensure_binary(span.service or "unnamed-python-service"))).strip("="),
+                str(sampling_priority),
+                str(sampling_mechanism.value),
+                # Float string rounded to 4 decimal places if set, otherwise an empty string
+                "{:0.4f}".format(sample_rate) if sample_rate is not None else "",
+            ]
+        )
+
+        # If the sampling decision hasn't changed, do not add an entry
+        # For example, if someone does `span.set_tag(MANUAL_KEEP_KEY)` multiple times in one trace
+        if self._last_upstream_service_entry == entry:
+            return
+
+        # Fetch any existing value (extracted from propagation headers, or we have already set ourselves)
+        existing = self._meta.get(UPSTREAM_SERVICES_KEY)
+
+        # We have an existing header value and a previously set upstream service entry by us
```

Comment:
You have it above but we should say here that any previous entry set by the same instrumented service would be replaced.